msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java

```diff
@@ -884,71 +887,78 @@ public DiscussionTopic createDiscussionForumTopic(DiscussionForum forum) {
     }
     
     
-    public void saveDiscussionForumTopic(DiscussionTopic topic) {	
-    	saveDiscussionForumTopic(topic, false);
+    public DiscussionTopic saveDiscussionForumTopic(DiscussionTopic topic) {
+    	return saveDiscussionForumTopic(topic, false);
     }
 
     /**
      * Save a discussion forum topic
      */
-    public void saveDiscussionForumTopic(DiscussionTopic topic, boolean parentForumDraftStatus) {
-    	saveDiscussionForumTopic(topic, parentForumDraftStatus, getCurrentUser(), true);
+    public DiscussionTopic saveDiscussionForumTopic(DiscussionTopic topic, boolean parentForumDraftStatus) {
+    	return saveDiscussionForumTopic(topic, parentForumDraftStatus, getCurrentUser(), true);
     }
     
-    public void saveDiscussionForumTopic(DiscussionTopic topic, boolean parentForumDraftStatus, String currentUser, boolean logEvent) {
+    public DiscussionTopic saveDiscussionForumTopic(DiscussionTopic topic, boolean parentForumDraftStatus, String currentUser, boolean logEvent) {
         boolean isNew = topic.getId() == null;
 
+        topic.setModified(new Date());
+
+        transformNullsToFalse(topic, currentUser);
+
+        //make sure availability is set properly
+        topic.setAvailability(ForumScheduleNotificationCover.makeAvailableHelper(topic.getAvailabilityRestricted(), topic.getOpenDate(), topic.getCloseDate()));
+
+        DiscussionTopic topicReturn = topic;
+        if (isNew) {
+            DiscussionForum discussionForum = (DiscussionForum) getForumByIdWithTopics(topic.getBaseForum().getId());
+            discussionForum.addTopic(topic);
+        } else {
+            topicReturn = getHibernateTemplate().merge(topic);
```

Comment:
Given that these services are already advised by spring tx lets use this pattern:
`getSessionFactory().getCurrentSession().merge(topic);`

Reply:
eventually we want to remove HibernateTemplate as it's a deprecated pattern.

Reply:
Done!